### PR TITLE
Security and data type changes for dev local coordinator mode

### PR DESCRIPTION
**Pull request by tanjialiang**
This reduces the time developer has to hack the codebase in order to bring up a local presto cluster with cpp worker.

Test plan 
* Run local coordinator with new code; Launched 1 cpp worker with corresponding discovery uri; Ran a select count(*) query on orders table in hive catalog and tpch schema. Successfully got result back.

```
== NO RELEASE NOTE ==
```

### CODE REVIEW

File: presto-hive/src/test/java/com/facebook/presto/hive/HiveQueryRunner.java [hunk 1]
```diff
@@ -136,7 +136,7 @@ public static DistributedQueryRunner createQueryRunner(
 
         DistributedQueryRunner queryRunner =
                 DistributedQueryRunner.builder(createSession(Optional.of(new SelectedRole(ROLE, Optional.of("admin")))))
-                        .setNodeCount(workerCount.orElse(4))
+                        .setNodeCount(workerCount.orElse(1))
```
Comment (highker) [inline presto-hive/src/test/java/com/facebook/presto/hive/HiveQueryRunner.java:139]: Let's leave this part untouched as Hive query runner will be used by a lot of integration tests. We will need parallelism coverage

File: presto-hive/src/test/java/com/facebook/presto/hive/TestHiveExternalWorkersQueries.java [hunk 2]
```diff
@@ -228,4 +239,51 @@ public void testValues()
     {
         assertQuery("SELECT 1, 0.24, ceil(4.5), 'A not too short ASCII string'");
     }
+
+    public static void main(String[] args)
```
Comment (highker) [inline presto-hive/src/test/java/com/facebook/presto/hive/TestHiveExternalWorkersQueries.java:243]: Can we separate out this into another new query runner? Maybe a new class called `HiveExternalWorkerQueryRunner`. Then use this new class for (1) your local laptop development and (2) as a util to be used in this test (`TestHiveExternalWorkersQueries.java`)

**Comment by tanjialiang** [comment]
Addressed all comments

File: presto-hive/src/test/java/com/facebook/presto/hive/HiveExternalWorkerQueryRunner.java [hunk 3]
```diff
@@ -0,0 +1,176 @@
+package com.facebook.presto.hive;
```
Comment (highker) [inline presto-hive/src/test/java/com/facebook/presto/hive/HiveExternalWorkerQueryRunner.java:1]: The class misses apache license header. Check the one from other classes and copy it here.

File: presto-hive/src/test/java/com/facebook/presto/hive/HiveExternalWorkerQueryRunner.java [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+        String prestoServerPath = System.getenv("PRESTO_SERVER");
+        String baseDataDir = System.getenv("DATA_DIR");
+        String workerCount = System.getenv("WORKER_COUNT");
+
+        return createQueryRunner(Optional.ofNullable(prestoServerPath),
+                Optional.ofNullable(baseDataDir).map(Paths::get),
+                Optional.ofNullable(workerCount).map(Integer::parseInt));
+    }
+
+    public static DistributedQueryRunner createQueryRunner(Optional<String> prestoServerPath,
```
Comment (highker) [inline presto-hive/src/test/java/com/facebook/presto/hive/HiveExternalWorkerQueryRunner.java:39]: nit: move the first param (`Optional<String> prestoServerPath`) to its own line.

File: presto-hive/src/test/java/com/facebook/presto/hive/HiveExternalWorkerQueryRunner.java [hunk 5]
```diff
@@ -0,0 +1,176 @@
...
+    private static final Logger log = Logger.get(HiveQueryRunner.class);
+
+    public static QueryRunner createQueryRunner()
+            throws Exception
+    {
+        String prestoServerPath = System.getenv("PRESTO_SERVER");
+        String baseDataDir = System.getenv("DATA_DIR");
+        String workerCount = System.getenv("WORKER_COUNT");
+
+        return createQueryRunner(Optional.ofNullable(prestoServerPath),
```
Comment (highker) [inline presto-hive/src/test/java/com/facebook/presto/hive/HiveExternalWorkerQueryRunner.java:34]: nit: move the first param `Optional.ofNullable(prestoServerPath)` to its own line.

File: presto-hive/src/test/java/com/facebook/presto/hive/HiveExternalWorkerQueryRunner.java [hunk 6]
```diff
@@ -0,0 +1,176 @@
...
+
+        if (!prestoServerPath.isPresent()) {
+            return defaultQueryRunner;
+        }
+
+        defaultQueryRunner.close();
+
+        // Make query runner with external workers for tests
+        DistributedQueryRunner queryRunner = HiveQueryRunner.createQueryRunner(ImmutableList.of(),
+                ImmutableMap.of("optimizer.optimize-hash-generation", "false",
```
Comment (highker) [inline presto-hive/src/test/java/com/facebook/presto/hive/HiveExternalWorkerQueryRunner.java:67]: move `"optimizer.optimize-hash-generation", "false"` to its own line

File: presto-hive/src/test/java/com/facebook/presto/hive/HiveExternalWorkerQueryRunner.java [hunk 7]
```diff
@@ -0,0 +1,176 @@
...
+                    baseDataDirFile = baseDataDirFile.getParentFile();
+                }
+                if (!baseDataDirFile.canRead() || !baseDataDirFile.canWrite()) {
+                    log.error("Error: The ancestor directory " + baseDataDirFile.getAbsolutePath() + " is not readable/writable.");
+                    System.exit(1);
+                }
+            }
+
+            baseDataDir = Optional.of(baseDataDirFile.toPath());
+        }
```
Comment (highker) [inline presto-hive/src/test/java/com/facebook/presto/hive/HiveExternalWorkerQueryRunner.java:168]: Let's remove this. I don't think it is used.

File: presto-hive/src/test/java/com/facebook/presto/hive/HiveExternalWorkerQueryRunner.java [hunk 8]
```diff
@@ -0,0 +1,176 @@
...
+        createOrders(defaultQueryRunner);
+
+        if (!prestoServerPath.isPresent()) {
+            return defaultQueryRunner;
+        }
+
+        defaultQueryRunner.close();
+
+        // Make query runner with external workers for tests
+        DistributedQueryRunner queryRunner = HiveQueryRunner.createQueryRunner(ImmutableList.of(),
```
Comment (highker) [inline presto-hive/src/test/java/com/facebook/presto/hive/HiveExternalWorkerQueryRunner.java:66]: nit: not need to create another variable `DistributedQueryRunner queryRunner = ...`. Directly return the result.

File: presto-hive/src/test/java/com/facebook/presto/hive/HiveQueryRunner.java (hunk 1 above)
Comment (highker) [reply presto-hive/src/test/java/com/facebook/presto/hive/HiveQueryRunner.java:139]: Forgot to put it back?

**Review by highker — COMMENTED** [review]
Could you squash all the commits into one? For the commit message, it would be nice to have title and message body to have max 72 characters per line. https://chris.beams.io/posts/git-commit/ is a good read on writing commit message.

Overall LGTM; nits only.

**Comment by tanjialiang** [comment]
> Could you squash all the commits into one? For the commit message, it would be nice to have title and message body to have max 72 characters per line. https://chris.beams.io/posts/git-commit/ is a good read on writing commit message.
> 
> 
> 
> Overall LGTM; nits only.

Yeah let me try. I didn't even realize the commit thing. Took it for granted as if working with mercurial..

File: presto-hive/src/test/java/com/facebook/presto/hive/HiveQueryRunner.java (hunk 1 above)
Comment (tanjialiang) [reply presto-hive/src/test/java/com/facebook/presto/hive/HiveQueryRunner.java:139]: Interesting I remembered having changed it. Maybe somewhere in git operations reverted it back.

File: presto-hive/src/test/java/com/facebook/presto/hive/HiveExternalWorkerQueryRunner.java (hunk 5 above)
Comment (tanjialiang) [reply presto-hive/src/test/java/com/facebook/presto/hive/HiveExternalWorkerQueryRunner.java:34]: Are there any auto formatting plugin in intellij that can do this?

**Pull request closed without merging**